Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool.swift

```diff
@@ -12,8 +12,114 @@
 //
 //===----------------------------------------------------------------------===//
 
+import NIO
+
 enum HTTPConnectionPool {
-    struct Connection {
+    struct Connection: Equatable {
         typealias ID = Int
+
+        private enum Reference {
+//            case http1_1(HTTP1Connection)
+
+            #if DEBUG
+                case testing(ID, EventLoop)
+            #endif
+        }
+
+        private let _ref: Reference
+
+//        fileprivate static func http1_1(_ conn: HTTP1Connection) -> Self {
+//            Connection(_ref: .http1_1(conn))
+//        }
+
+        #if DEBUG
+            static func testing(id: ID, eventLoop: EventLoop) -> Self {
+                Connection(_ref: .testing(id, eventLoop))
+            }
+        #endif
+
+        var id: ID {
+            switch self._ref {
+//            case .http1_1(let connection):
+//                return connection.id
+            #if DEBUG
+                case .testing(let id, _):
+                    return id
+            #endif
+            }
+        }
+
+        var eventLoop: EventLoop {
+            switch self._ref {
+//            case .http1_1(let connection):
+//                return connection.channel.eventLoop
+            #if DEBUG
+                case .testing(_, let eventLoop):
+                    return eventLoop
+            #endif
+            }
+        }
+
+        #if DEBUG
```

Comment:
Don't hide this behind debug: use underscores and `testOnly` (e.g. `__testOnly_channel`.)